lesson05/shapes/circle.go

```diff
@@ -0,0 +1,50 @@
+package shapes
+
+import (
+	"fmt"
+	"math"
+)
+
+type Circle struct {
+	radius float64
+}
+
+func (c Circle) Area() (float64, bool) {
```

Comment:
This is not Go idiomatic. The second return value should be error, which needs to be returned on incorrect input (negative radius value in this case). If the error is not nil, the float64 value needs to be default. Same with other methods on Circle and Rectangle

Reply:
fixed